students/mariusz_michalczyk/lesson_08_regex/email_validator.py

```diff
@@ -0,0 +1,15 @@
+import re
+
+
+def validate_email(email):
+    rgx_email = re.compile(r'[a-zA-Z0-9._%+-]+@[A-Za-z0-9]+(\.[a-zA-Z]{2,4})')
+    matched = rgx_email.search(email)
```

Comment:
`validate_email('####valid@email.com###')` will pass, because you used `search` instead of `match` (alternatively you didn't prepend the regex with `^`). The invalid characters at the end will also match with `match`, you need `$` at the end.